src/shared/lib/user-benefits-api.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from 'zod';
+
+// list of benefits we're looking for
+const BenefitEnum = z.enum([
+	'adFree',
+	'allowRejectAll',
+	'hideSupportMessaging',
+]);
+
+export const userBenefitsResponseSchema = z.object({
+	benefits: z
+		.array(z.string()) // Accept any array of strings
+		.transform((values) =>
+			values.filter((val) => BenefitEnum.safeParse(val).success),
```

Comment:
```suggestion
			values.filter((val): val is z.infer<typeof BenefitEnum> => BenefitEnum.safeParse(val).success))
```

The way you've implemented it wont actually make the type correct, it is still being parsed as `z.string()`. Using the suggestion I have parses the type correctly to

```ts
type ResponseType = { benefits: ("adFree" | "allowRejectAll" | "hideSupportMessaging")[]; }
```